pymic/offload_library.py

```diff
@@ -65,16 +65,19 @@ def _check_k1om(library):
 
     @staticmethod
     def _find_library(library):
-        for path in config._search_path.split(":"):
-            try:
-                files = [f for f in os.listdir(path)
-                         if os.path.isfile(os.path.join(path, f))]
-                if library in files:
-                    filename = os.path.join(path, library)
-                    if OffloadLibrary._check_k1om(filename):
-                        return filename
-            except:
-                pass
+        if os.path.isabs(library):
+            abspath = library
+        else:
+            for path in config._search_path.split(':'):
+                abspath = os.path.join(path, library)
+
+                if os.path.isfile(abspath):
+                    break
+            else:
+                return
+
+        if OffloadLibrary._check_k1om(abspath):
+            return abspath
```

Comment:
The original code went through the search path and made sure that if a name match was found, but the library wasn't for MIC.  This code bails out from the search for the first library found by name. Can you please update the patch to revert this behavior?